msf_core/include/msf_core/implementation/msf_measurement_inl.h

```diff
@@ -101,7 +101,8 @@ void MSF_MeasurementBase<EKFState_T>::CalculateAndApplyCorrection(
 
   if(enable_mah_outlier_rejection_){
     //calculate mahalanobis distance
-    double mah_dist_squared = res_delayed.transpose() * S_inverse * res_delayed;
+    Eigen::MatrixXd mah_dist_squared_temp = res_delayed.transpose() * S_inverse * res_delayed;
+    double mah_dist_squared = mah_dist_squared_temp(0,0);
```

Comment:
Makes sense, I was actually a little surprised that it allowed the implicit conversion when I first did it